File: io-hotmoka-network/src/main/java/io/hotmoka/network/internal/models/requests/GameteCreationTransactionRequestModel.java
package io.hotmoka.network.internal.models.requests;

import java.math.BigInteger;

import io.hotmoka.beans.requests.GameteCreationTransactionRequest;
import io.hotmoka.network.internal.models.values.TransactionReferenceModel;

public class GameteCreationTransactionRequestModel extends InitialTransactionRequestModel {
    private BigInteger amount;
    private String publicKey;
    private TransactionReferenceModel classpath;

    public void setClasspath(TransactionReferenceModel classpath) {
        this.classpath = classpath;
    }

    public void setAmount(BigInteger amount) {
        this.amount = amount;
    }

    public void setPublicKey(String publicKey) {
        this.publicKey = publicKey;
    }

    public GameteCreationTransactionRequest toBean() {
    	return new GameteCreationTransactionRequest(classpath.toBean(), amount, publicKey);
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/requests/GameteCreationTransactionRequestModel.kt
package io.hotmoka.network.thin.client.models.requests

import io.hotmoka.network.thin.client.models.values.TransactionReferenceModel

class GameteCreationTransactionRequestModel(
    val initialAmount: String,
    val publicKey: String,
    val classpath: TransactionReferenceModel
): InitialTransactionRequestModel()